examples/nextjs-react18-editor-sample/src/components/map-edit-toolbox.tsx

```diff
@@ -0,0 +1,443 @@
+  margin: 0.05em;
+  padding: 0.1em 0.2em;
+  :hover {
+    background: rgb(128, 137, 133);
+  }
+`;
+
+const SubToolsContainer = styled.div`
+  position: relative;
+`;
+
+const SubTools = styled.div`
+  display: flex;
+  flex-direction: row-reverse;
+  position: absolute;
+  top: 0;
+  right: 0;
+`;
+
+export type Props = {
+  editMode: any;
+  selectMode: any;
+  selectedFeatureIndexes: any;
+  modeConfig: any;
+  geoJson: any;
+  onSetEditMode: (mode: any) => unknown;
+  onSetSelectMode: (mode: any) => unknown;
+  onSetSelectedFeatureIndexes: (mode: any) => unknown;
+  onSetModeConfig: (modeConfig: any) => unknown;
+  onSetGeoJson: (geojson: any) => unknown;
+  onImport: (imported: any) => unknown;
+};
+const MODE_GROUPS: {
+  modes: {
+    mode: typeof ViewMode;
+    content: JSX.Element;
+    title?: string;
+  }[];
+}[] = [
+    {
+      modes: [{ mode: DrawPointMode, content: <MapPinIcon /> }],
+    },
+    {
+      modes: [
+        {
+          mode: DrawLineStringMode,
+          content: <ActivityIcon />,
+        },
+      ],
+    },
+    {
+      modes: [
+        { mode: DrawPolygonMode, content: <HexagonIcon /> },
+        { mode: DrawRectangleMode, content: <RectangleHorizontalIcon /> },
+        { mode: DrawCircleFromCenterMode, content: <CircleIcon /> },
+      ],
+    },
+    {
+      modes: [
+        { mode: MeasureDistanceMode, content: <RulerIcon /> },
+        { mode: MeasureAngleMode, content: <ChevronLeftIcon /> },
+        { mode: MeasureAreaMode, content: <TriangleIcon /> },
+      ],
+    },
+  ];
+
+function ModeButton({ buttonConfig, mode, onClick }: any) {
+  return (
+    <Button
+      title={buttonConfig.title || buttonConfig.mode.name}
+      active={buttonConfig.mode === mode}
+      onClick={onClick}
+    >
+      {buttonConfig.content}
+    </Button>
+  );
+}
+function ModeGroupButtons({ modeGroup, mode, onSetMode }: any) {
+  const [expanded, setExpanded] = React.useState(false);
+
+  const { modes } = modeGroup;
+
+  let subTools = null;
+
+  if (expanded) {
+    subTools = (
+      <SubTools>
+        {modes.map(
+          (
+            buttonConfig: { mode: any; content: any; title: string },
+            i: any
+          ) => (
+            <ModeButton
+              key={i}
+              buttonConfig={buttonConfig}
+              mode={mode}
+              onClick={() => {
+                onSetMode(() => buttonConfig.mode);
+                setExpanded(false);
+              }}
+            />
+          )
+        )}
+      </SubTools>
+    );
+  }
+
+  const buttonConfig = modes.find((m: any) => m.mode === mode) || modes[0];
+
+  return (
+    <SubToolsContainer>
+      {subTools}
+      <ModeButton
+        buttonConfig={buttonConfig}
+        mode={mode}
+        onClick={() => {
+          onSetMode(() => buttonConfig.mode);
+          setExpanded(true);
+        }}
+      />
+    </SubToolsContainer>
+  );
+}
+
+const BlackJson = { type: "FeatureCollection", features: [] };
```

Comment:
It is not super obvious what `BlackJson` means without going through the code. Maybe TSDoc style comments in front of global constants and functions?

Reply:
got it,I added some comments